src/lib/db.ts

```diff
@@ -0,0 +1,83 @@
+import * as firebase from "firebase";
+import { AppMode } from "../models/stores";
+
+export type IDBConnectOptions = IDBAuthConnectOptions | IDBNonAuthConnectOptions;
+export interface IDBAuthConnectOptions {
+  appMode: "authed";
+  rawFirebaseToken: string;
+}
+export interface IDBNonAuthConnectOptions {
+  appMode: "dev" | "test";
+}
+
+export class DB {
+  private appMode: AppMode;
+  private firebaseUser: firebase.User | null = null;
+
+  public get isConnected() {
+    return this.firebaseUser !== null;
+  }
+
+  public connect(options: IDBConnectOptions) {
+    return new Promise<boolean>((resolve, reject) => {
+      if (this.isConnected) {
+        reject("Already connected to database!");
```

Comment:
Is rejecting preferable in this case? Logically, it would seem one could just as easily resolve on the grounds that the desired condition is already met.

Reply:
I added it here to ensure we are getting a new connection per test.  We will be only connecting to this in one place in the code.